src/shiplog.c

```diff
@@ -669,3 +671,21 @@ void shiplog_listLog( int logid, char *type,int *nentries, char ***logentries, i
    *nentries = n;
    
 }
+
+/*
+ * @brief Checks to see if the log family exists
+ *
+ * @param idstr ID string for the log family
+ */
+int shiplog_getID( const char *idstr )
+{
+   int id = -1;
+   int i;
+   for ( i=0; i<shipLog->nlogs; i++ ) {
+      if ( (shipLog->idstrList[i]==NULL && idstr==NULL) || (shipLog->idstrList[i]!=NULL && idstr!=NULL && !strcmp( idstr, shipLog->idstrList[i] ) ) ){
```

Comment:
This line is too long, stuff is squished together, and there should be parentheses around the checks between && ops. This is how I broke it up:

```
      if ( ( ( shipLog->idstrList[i] == NULL ) && ( idstr == NULL) )
            || ( ( shipLog->idstrList[i] != NULL ) && ( idstr != NULL )
               && ( strcmp(idstr, shipLog->idstrList[i]) == 0 ) ) ) {
```